airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/task/DestinationTaskLauncherTest.kt

```diff
@@ -65,10 +66,13 @@ class DestinationTaskLauncherTest {
     class MockSetupTaskFactory : SetupTaskFactory {
         val hasRun: Channel<Unit> = Channel(Channel.UNLIMITED)
 
-        override fun make(taskLauncher: DestinationTaskLauncher): SetupTask {
+        override fun make(
+            taskLauncher: DestinationTaskLauncher,
+        ): SetupTask {
             return object : SetupTask {
                 override suspend fun execute() {
                     hasRun.send(Unit)
+                    throw RuntimeException("Setup task failed")
```

Comment:
we should probably split these out eventually :P but definitely fair to leave this as-is for now.

Reply:
Actually, the smell here is because
* the exception handler is a separate interface
* BUT the task launcher is handling building the wrappers and directing traffic

I fixed that by moving the wrapper stuff into the interface and changing its core contract to just `fun <T: Task> withExceptionHandling(task: T): Task`. So the launcher's job is just to wrap whatever task it wants handling for. That separates the concerns better and obviates the test smell.